md2conf/api.py

```diff
@@ -328,6 +328,8 @@ def update_page(self, page_id: str, new_content: str) -> None:
         page = self.get_page(page_id)
 
         try:
+            # sanitize_confluence throws an error for unknown characters. Which if not fix will always result in a page update.
+            # as old_content != new_content, since confluence will not save unknown characters.
```

Comment:
From what I recall, this was with UTF-8 characters that are not in ASCII, which are perfectly fine in HTML (and Confluence XHTML) but appear to cause an issue with the `lxml` parser. I think if we do a first pass on the document, and replace these characters, we can avoid the infinite update loop.

Reply:
Some of our markdown had characters such as “ instead of ", likely the cause of copy paste. What would happen is that content of first upload would be published as &rsquo; in confluence. And then upon running the tool again, sanitize would throw. The result would be the page would regardless of changes would always be updated.

Reply:
Yeah, this is because `lxml` by default doesn't recognize HTML entities (character escapes like `&rsquo;`). I gave this a shot earlier but it turned out to require more than the 30 minutes I had at that time. Once HTML entities are properly handled, `sanitize` shouldn't have any issues processing documents with entities in them.

Reply:
I am happy to remove the comment in favor of an issue if you like.

Reply:
This issue has been addressed on the mainline branch `master` by adding an entity definition document `entity.dtd`.

Reply:
Thanks, I've resync'd and it works well!